worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -282,13 +282,21 @@ function isLastAction(action) {
 }
 
 function getCurrentWorkerName(document) {
-    return document.getApplication().getService(com.hpe.caf.api.ConfigurationSource.class)
-            .getConfiguration(com.hpe.caf.worker.document.config.DocumentWorkerConfiguration.class).getWorkerName();
+    try {
+        return document.getApplication().getClass().getMethod("getName").invoke(document.getApplication());
+    } catch (e) {
+        return document.getApplication().getService(com.hpe.caf.api.ConfigurationSource.class)
+                .getConfiguration(com.hpe.caf.worker.document.config.DocumentWorkerConfiguration.class).getWorkerName();
+    }
```

Comment:
Don't use reflection unnecessarily:

    return document.getApplication().getClass().getMethod("getName").invoke(document.getApplication());

can be just:

    return document.application.name;

Reply:
Also you can probably check if it is `undefined` and use that to know whether or not to fallback to the original code.

Reply:
if the method getName or getVersion is not present, it was throwing an abstractMethodError:null.
i.e 
var workerName = document.application.name; ---> threw exception.
I had to use the catch block to fallback on the original option.

Reply:
You could tidy this up using the `||` operator.

    return document.application.name
        || document.application.getService(com.hpe.caf.api.ConfigurationSource.class)
                               .getConfiguration(com.hpe.caf.worker.document.config.DocumentWorkerConfiguration.class)
                               .getWorkerName();